src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/FeatureFlag/FeatureFlagsFormDataProvider.php

```diff
@@ -110,6 +121,7 @@ public function setData(array $flagsData)
         }
 
         $this->doctrineEntityManager->flush();
+        $this->routeProvider->getLegacyRoutes();
```

Comment:
Is this to force clearing the cache? If so at least you should add a comment to understand WTF this is doing here 😛

But I'm not sure this is enough to clear the cache right? Since the cached service is injected it will always return a value based on the cache anyway no? I was more thinking that you'd need the clear the cache by getting the cached item like here:

https://github.com/PrestaShop/PrestaShop/blob/3d06748975e00c548e940bf82862d9d211ae94d4/src/PrestaShopBundle/Routing/Converter/CacheProvider.php#L72-L80

So you can then manually clear it accurately, this will avoid clearing the whole cache.